# Week 7: Groupby
wil be looking at the drinks dataset again :)

In [1]:
import pandas as pd

In [2]:
drinks_df = pd.read_csv('https://bit.ly/drinksbycountry')

#### basic information about the dataset

In [3]:
drinks_df.head()

country  beer_servings  spirit_servings  wine_servings  \
0  Afghanistan              0                0              0   
1      Albania             89              132             54   
2      Algeria             25                0             14   
3      Andorra            245              138            312   
4       Angola            217               57             45   

   total_litres_of_pure_alcohol continent  
0                           0.0      Asia  
1                           4.9    Europe  
2                           0.7    Africa  
3                          12.4    Europe  
4                           5.9    Africa

In [4]:
drinks_df.tail()

country  beer_servings  spirit_servings  wine_servings  \
188  Venezuela            333              100              3   
189    Vietnam            111                2              1   
190      Yemen              6                0              0   
191     Zambia             32               19              4   
192   Zimbabwe             64               18              4   

     total_litres_of_pure_alcohol      continent  
188                           7.7  South America  
189                           2.0           Asia  
190                           0.1           Asia  
191                           2.5         Africa  
192                           4.7         Africa

##### = data is listed in alphabetical order of countries

In [5]:
len(drinks_df)

193

##### = 193 countries in the dataset

#### let's explore the dataset

##### 1. total pure alcohol servings in each continent

In [6]:
drinks_df.groupby(['continent'])['total_litres_of_pure_alcohol'].sum()

continent
Africa           159.4
Asia              95.5
Europe           387.8
North America    137.9
Oceania           54.1
South America     75.7
Name: total_litres_of_pure_alcohol, dtype: float64

##### 2. filter for Europe

In [7]:
europe_df = drinks_df[drinks_df['continent'] == "Europe"]

##### which countries consume the most pure alcohol?

In [8]:
europe_df.sort_values('total_litres_of_pure_alcohol', ascending=False).head()

country  beer_servings  spirit_servings  wine_servings  \
15         Belarus            142              373             42   
98       Lithuania            343              244             56   
3          Andorra            245              138            312   
45  Czech Republic            361              170            134   
61          France            127              151            370   

    total_litres_of_pure_alcohol continent  
15                          14.4    Europe  
98                          12.9    Europe  
3                           12.4    Europe  
45                          11.8    Europe  
61                          11.8    Europe

##### 3. which countries in Asia drink the most wine?

In [14]:
asia_df = drinks_df[drinks_df['continent'] == "Asia"]

In [15]:
asia_df.sort_values('wine_servings', ascending=False).head()

country  beer_servings  spirit_servings  wine_servings  \
92                 Laos             62                0            123   
141  Russian Federation            247              326             73   
177        Turkmenistan             19               71             32   
94              Lebanon             20               55             31   
167               Syria              5               35             16   

     total_litres_of_pure_alcohol continent  
92                            6.2      Asia  
141                          11.5      Asia  
177                           2.2      Asia  
94                            1.9      Asia  
167                           1.0      Asia

##### 4. min, max, count, mean for all continents for pure alcohol

In [16]:
drinks_df.groupby(['continent'])['total_litres_of_pure_alcohol'].agg(['min', 'max', 'count', 'mean']).reset_index()

continent  min   max  count      mean
0         Africa  0.0   9.1     53  3.007547
1           Asia  0.0  11.5     44  2.170455
2         Europe  0.0  14.4     45  8.617778
3  North America  2.2  11.9     23  5.995652
4        Oceania  0.0  10.4     16  3.381250
5  South America  3.8   8.3     12  6.308333

##### 5. min, max, count, mean for Asian countries for pure alcohol

In [18]:
asia_df = drinks_df[drinks_df['continent'] == "Asia"]

In [19]:
asia_df.groupby['total_litres_of_pure_alcohol'].agg(['min', 'max', 'count', 'mean']).reset_index()

TypeError: 'method' object is not subscriptable